### ISSUE TITLE
Puppet.newtype is deprecated

### ISSUE THREAD

**Issue by TravellingGuy** [ffrank/puppet-constraints#2]
Thanks for the constraints module. Solved a lot of problems for me!

I'm using it with puppet 4.2 and when I do an agent run, I get this warning:

```
Warning: Creating constraint via Puppet.newtype is deprecated and will be removed in a future release. Use Puppet::Type.newtype instead.
   (at /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb:145:in `newtype')
```

Being just a warning, it still runs.

**Comment by igalic** [comment]
@TravellingGuy would you like to put together a patch for this?

**Comment by TravellingGuy** [comment]
@igalic Pull request sent.

**Comment by ffrank** [comment]
Sorry for the delays - I had literally zero time to run a few Puppet version tests. "Soon"!

**Comment by ffrank** [comment]
I just released version `0.2.2` on the Forge, including your fix. I believe that the issue is now resolved.

Thanks for reporting and fixing this!
